File: Samples/JotFormTest/Classes/Controller/GetAllFormsViewController.m
//
//  GetAllFormsViewController.m
//  JotFormTest
//
//  Created by Interlogy, LLC on 1/6/14.
//  Copyright (c) 2014 Interlogy, LLC. All rights reserved.
//

#import "GetAllFormsViewController.h"
#import "DataListViewController.h"
#import "SharedData.h"
#import "SVProgressHUD.h"
#import <JotForm_iOS/JotForm.h>

@interface GetAllFormsViewController () {
           NSArray *orderbyList;
           SharedData *sharedData;
}

@property (nonatomic,weak) IBOutlet UITextField *offsetTextField;
@property (nonatomic,weak) IBOutlet UITextField *limitTextField;
@property (nonatomic,weak) IBOutlet UIPickerView *pickerView;
@property (nonatomic,weak) IBOutlet UITextField *filterTextField;
@property (nonatomic,weak) IBOutlet UIBarButtonItem *getBarButtonItem;

@end

@implementation GetAllFormsViewController

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view from its nib.
    
    [self initData];
    [self initUI];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - user definition method

- (void)initUI
{
    self.title = @"Get all forms";
    self.navigationItem.rightBarButtonItem = self.getBarButtonItem;
}

- (void)initData
{
    orderbyList = [[SharedData sharedData] getFormOrderbyList];
}

- (void)loadForms
{
    [SVProgressHUD showWithStatus:@"Loading forms..."];
    
    NSInteger offset = 0;
    NSInteger limit = 0;
    
    if (self.offsetTextField.text.length > 0) {
        offset = [self.offsetTextField.text integerValue];
    }
    
    if (self.limitTextField.text.length > 0) {
        limit = [self.limitTextField.text integerValue];
    }
    
    NSString *orderby = [orderbyList objectAtIndex:[self.pickerView selectedRowInComponent:0]];
    
    [[SharedData sharedData].apiClient getForms:offset limit:limit orderBy:orderby filter:nil onSuccess:^(id result){
       
        [SVProgressHUD dismiss];
        
        if (result) {
            NSInteger responseCode = [[result objectForKey:@"responseCode"] integerValue];
            
            if (responseCode == 200 || responseCode == 206) {
                NSArray *formsArray = [result objectForKey:@"content"];
                [self startDataListViewController:formsArray];
            }
        }
  
    } onFailure:^(id error) {
        [SVProgressHUD dismiss];
    }];
}

- (void)startDataListViewController : (NSArray *) datalist
{
    DataListViewController *dataListVc = [[DataListViewController alloc] initWithNibName:@"DataListViewController" bundle:nil];

    [self.navigationController pushViewController:dataListVc animated:YES];
    
    [dataListVc setList:datalist type:DataListTypeFormList];
}

#pragma mark IBAction

- (IBAction)getFormsButtonClicked : (id) sender {
       [self loadForms];
}

#pragma mark UIPickerViewDataSource

- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
    return 1;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
    return [orderbyList count];
}

#pragma mark UIPickerView delegate

- (NSString *)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component {
    NSString *rowStr = [orderbyList objectAtIndex:row];
    
    return rowStr;
}

@end

File: Samples/JotFormTest/Classes/Controller/GetAllFormsViewController.swift
//
//  GetAllFormsViewController.swift
//  JotFormTest
//
//  Created by Curtis Stilwell on 2/1/18.
//  Copyright © 2018 wang. All rights reserved.
//

import Foundation

class GetAllFormsViewController: UIViewController ,UIPickerViewDelegate, UIPickerViewDataSource {
    var orderbyList = [Any]()
    var sharedData: SharedData?
    
    @IBOutlet weak var offsetTextField: UITextField!
    @IBOutlet weak var limitTextField: UITextField!
    @IBOutlet weak var pickerView: UIPickerView!
    @IBOutlet weak var filterTextField: UITextField!
    @IBOutlet weak var getBarButtonItem: UIBarButtonItem!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view from its nib.
        initData()
        initUI()
    }
    
    func initUI() {
        title = "Get all forms"
        navigationItem.rightBarButtonItem = getBarButtonItem
    }
    
    func initData() {
        orderbyList = SharedData().getFormOrderbyList()
    }
    
    func loadForms() {
        SVProgressHUD.show(withStatus: "Loading forms...")
        
        var offset: Int = 0
        var limit: Int = 0
       
        if offsetTextField.text!.count > 0 {
            offset = Int(offsetTextField.text!)!
        }
       
        if limitTextField.text!.count > 0 {
            limit = Int(limitTextField.text!)!
        }
        
        let orderby: String = orderbyList[pickerView.selectedRow(inComponent: 0)] as! String
        
        SharedData.sharedData.apiClient?.getForms(offset, limit: limit, orderBy: orderby, filter:nil, onSuccess: {(_ result: AnyObject) -> Void in
            SVProgressHUD.dismiss()
             let responseCode = result["responseCode"] as? Int
           
            if responseCode == 200 || responseCode == 206 {
                let formsArray = result["content"] as? [AnyObject]
                self.startDataListViewController(formsArray!)
            }
            
        }, onFailure: {(_ error: Any) -> Void in
            SVProgressHUD.dismiss()
        })
    }
    
    func startDataListViewController(_ datalist: [Any]) {
        let dataListVc = DataListViewController(nibName: "DataListViewController", bundle: nil)
        navigationController?.pushViewController(dataListVc, animated: true)
        dataListVc.setList(datalist as [AnyObject], type:DataListType.formList)
    }
    
    @IBAction func getFormsButtonClicked(_ sender: Any) {
        loadForms()
    }
    
    func numberOfComponents(in pickerView: UIPickerView) -> Int {
        return 1
    }
    
    func pickerView(_ pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return orderbyList.count
    }
    
    func pickerView(_ pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String? {
        let rowStr: String = orderbyList[row] as! String
        return rowStr
    }
}
